### PR TITLE
History

### PR DESCRIPTION
**Pull request by Fresheyeball**
History, Timer and Event Bindings.

### CODE REVIEW

File: Presentable.sublime-workspace [hunk 1]
```diff
@@ -80,60 +188,150 @@
 	"buffers":
 	[
 		{
-			"file": "tests/Chai/Chai.purs",
+			"file": "src/History/History.purs",
 			"settings":
 			{
-				"buffer_size": 861,
+				"buffer_size": 1873,
```
Comment (vrecan) [inline Presentable.sublime-workspace:194]: that's an oddly specific buffer size

File: src/History/History.purs [hunk 2]
```diff
@@ -1,61 +1,72 @@
...
+getData                     = unsafeForeignFunction [""] "window.history.state"
+getTitle                    = unsafeForeignFunction [""] "document.title"
+getUrl                      = unsafeForeignFunction [""] "location.pathname"
 
-------
+getState                    :: forall m d. (Monad m) => m (State d)
+getState                    = do d <- getData
+                                 t <- getTitle
+                                 u <- getUrl
+                                 return { title : t, url : u, "data" : d }
```
Comment (vrecan) [inline src/History/History.purs:42]: just curious, why id "data" a string while the others are obvious field names?

**Comment by vrecan** [comment]
overall I am happy that I can mostly follow what's going on

File: src/History/History.purs (hunk 2 above)
Comment (Fresheyeball) [inline src/History/History.purs:42]: I am trying to mimic the existing API

**Pull request merged**